### DISCUSSION TITLE
Blazor page reload

### DISCUSSION THREAD

**Discussion by lartusi** [dotnet/aspnetcore#57661] [Q&A]
I want to reload my blazor page without triggering a full browser refresh.
So invoking NavigationManager.NavigateTo(NavigationManager.Uri, true) is not an option.

I found a workaround navigating to a fake page and than back to the page i want:
```
string url = NavManager.Uri;
NavManager.NavigateTo("/fake-page", false); // Note: this is a dirty trick just to force the page reload
NavManager.NavigateTo(url, false);
```

Is there a smarter way to do the same?

**Comment by lucagaggero7** [comment]
try NavigateTo(put in parentheses the @page domain that your razor component does have usually at the top)

**Comment by SQL-MisterMagoo** [comment]
You would need to be more precise about what you want to happen  - "reload my blazor page" means what to you? 
Do you want to drop and recreate the instance of your "page" that is current?
Do you want to drop and recreate the DOM elements defined in your page?

What is the end result you want to achieve by "reload my blazor page"?

**Reply by lartusi** [reply]
I want to re-inizialite the whole page.
Same result as when you navigate to another page or when you navigate to the same page with the "forceLoad" option (but without full browser refresh).

I want to use client side routing like when you're navigating to another page.

I don't know if it's clear now.

**Reply by SQL-MisterMagoo** [reply]
AFAIK there is no way to do that without navigating away, but you could move the route to a new component that just has your existing page/component on it, then you can re-initialise by changing the `@key`...

Suppose your original page was this `MyPage.razor`
```html
@page "/mypage"

<some_markup>...</some_markup>
@code {
... some code
}
```

Then you could create a new component  `MyPageContent.razor` that is an exact copy of `MyPage.razor` without the `@page` attribute, then modify `MyPage.razor`  :
```html
@page "/mypage"

<MyPageContent @key=key OnReInitialize=NewKey />

@code {
  int key = 0;
  void NewKey() => key++;
}
```

In your new `MyPageContent.razor` add a new `EventCallback` called `OnReInitialize`
```csharp
[Property] public EventCallback OnReInitialize { get; set;  }
```

Now, whenever you want to "reload" you can `OnReInitialize.InvokeAsync()` from your `MyPageContent` component and it will be dropped and recreated.
